include/dali/core/mm/memory_kind.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020-2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_MM_MEMORY_KIND_H_
+#define DALI_CORE_MM_MEMORY_KIND_H_
+
+#include <type_traits>
+#include "dali/core/mm/memory_resource.h"
+
+namespace dali {
+namespace mm {
+
+/**
+ * @brief For run-time dispatch of memory kind - DO NOT USE unless REALLY necessary
+ */
```

Comment:
Maybe it would be a good idea to explain, why it shouldn't be used and what are the "REALLY necessary" circumstances?

Reply:
Will do.